src/dotty/tools/dotc/core/tasty/NameBuffer.scala

```diff
@@ -42,11 +42,13 @@ class NameBuffer extends TastyBuffer(10000) {
 
   private def withLength(op: => Unit): Unit = {
     val lengthAddr = currentAddr
-    writeByte(0)
+    val lengthFieldLength = 2
```

Comment:
A design goal of pickling is to be as compact as possible. Wasting one byte for each name is not compatible with that. We should find a scheme that needs the second byte only for long names. I would propose an optional parameter

    def withLength(op: => Unit, maxLength: Int = 127)

which gets specialized in `writeName`.